### PR TITLE
Fix stale/inconsistent values in update global concurrency limits model

### LINKED ISSUE #15377: Inactive global concurrency limits are shown as active when edited
### Bug summary

Given an inactive global concurrency limit, when I navigate to the concurrency page and table in the UI, it shows as inactive on the table but when I go to edit it, the modal shows it as active.

image

### Version info (`prefect version` output)

```Text
3.0.1
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by collincchoy**
This PR includes 3 fixes for the GCL(concurrency limits v2) update modal:
- Fix modal always initial shows active=true. Closes https://github.com/PrefectHQ/prefect/issues/15377
- Fix subsequent gcl updates show stale values in the modal. 
- Fix toggling active from the table doesn't update the value on the modal.

### CODE REVIEW

File: src/components/ConcurrencyLimitsV2UpdateModal.vue [hunk 1]
```diff
@@ -63,34 +60,23 @@
...
 
-  const active = ref(true)
-
-  const activeSlots = ref(props.concurrencyLimit.activeSlots)
-
-
-  const decay = ref(props.concurrencyLimit.slotDecayPerSecond)
-
-
-  const internalShowModal = computed({
```
Comment (collincchoy) [inline src/components/ConcurrencyLimitsV2UpdateModal.vue:74]: moved this to defineModel.

for funzies.

File: src/components/ConcurrencyLimitsV2UpdateModal.vue [hunk 2]
```diff
@@ -63,34 +60,23 @@
...
   const reset = (): void => {
     name.value = props.concurrencyLimit.name
     limit.value = props.concurrencyLimit.limit
-    decay.value = props.concurrencyLimit.slotDecayPerSecond ?? 0
     active.value = props.concurrencyLimit.active ?? true
     activeSlots.value = props.concurrencyLimit.activeSlots ?? 0
+    decay.value = props.concurrencyLimit.slotDecayPerSecond ?? 0
   }
 
+  watch(() => props.concurrencyLimit, reset)
```
Comment (collincchoy) [inline src/components/ConcurrencyLimitsV2UpdateModal.vue:78]: reset the form fields when the prop is updated - i.e. update the limit from the table(flip active toggle) and then open the modal

File: src/components/ConcurrencyLimitsV2UpdateModal.vue [hunk 3]
```diff
@@ -100,8 +86,8 @@
           name: name.value,
           limit: limit.value,
           active: active.value,
-          slotDecayPerSecond: decay.value,
           activeSlots: activeSlots.value,
+          slotDecayPerSecond: decay.value,
```
Comment (collincchoy) [inline src/components/ConcurrencyLimitsV2UpdateModal.vue:90]: using consistent ordering as elsewhere in the file

File: src/components/ConcurrencyLimitsV2UpdateModal.vue [hunk 4]
```diff
@@ -63,34 +60,23 @@
     isGreaterThanZeroOrNull,
   ])
 
-  const active = ref(true)
```
Comment (collincchoy) [inline src/components/ConcurrencyLimitsV2UpdateModal.vue:66]: this aint' quite right. if it's already active.

**Review by znicholasbrown — APPROVED** [review]
this lgtm but @jeanluciano might be a better reviewer

**Review by abrookins — APPROVED** [review]
Whoa a nice omnibus of fixes here @collincchoy. 👍

**Pull request merged**